workflow/scripts/structure_learning_algorithms/cdt.py

```diff
@@ -0,0 +1,82 @@
+import numpy as np
+import pandas as pd
+import networkx as nx
+from contextlib import contextmanager
+import time
+import signal
+
+@contextmanager
```

Comment:
It shouldn't be necessary to define the timeout functions here. Maybe have a look at [causallearn_graphs](https://github.com/felixleopoldo/benchpress/blob/master/workflow/rules/structure_learning_algorithms/causallearn_grasp/script.py#L6) to see how it is handled